service/dataset.py

```diff
@@ -48,23 +42,65 @@ def read(dataset_request):
     except:
         # TODO: Temp fix for testing - Do better cleanup if secret missing
         # dataset["dataset_type"]["token"] = {'name':dataset_name,'value':42}
-        pass 
-    
+        pass
+
+    return dataset
+
+def readreleased(dataset_request):
+    dataset = _read_helper(dataset_request)
+
     # Check/Decrement the budget before returning dataset
     adjusted_budget = dataset["budget"] - dataset_request["budget"]
     if adjusted_budget >= 0.0:
         dataset["budget"] = adjusted_budget
     else:
-        abort(412, "Not enough budget for read. Remaining budget: {}".format(dataset_name))
+        abort(412, "Not enough budget for read. Remaining budget: {}".format(dataset["budget"]))
+    
+    return dataset
+
+def release(release_request):
+    dataset_name = release_request["dataset_name"]
+
+    if dataset_name not in DATASETS:
+        abort(401, "Dataset id {} doesn't exist.".format(dataset_name))
+
+    # Check secret here, to make sure this comes from service
+
+    dataset = DATASETS[dataset_name]
```

Comment:
I like how sleek this looks but I think release will be more complicated.

release is effectively a new dataset.register of a released dataset. The name should be unique among released datasets(like datasets) and the new dataset will probably be created within a module so renaming/exposing a previously private dataset as a released dataset won't work.

Client:
- submits module

Module:
 - reads private data in
 - creates something from it that is differentially private
 - registers a new Dataset, ReleasedDataset, that points to this new type of dataset that costs budget to read but is not private.

This means Modules will need to create a local csv(for testing) or a local csv they upload as a dataset file to dataverse and then register that dataset as a released dataset. There may be other sources we support later but those are the two valid ones now. 

No need to work on uploading a dataverse dataset for this PR, adding a local_csv should suffice

Reply:
Module has been updated according to this spec